.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+name: CI
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+env:
+  app: sidekiq_publisher
+
+jobs:
+  lint:
+    name: Lint (Ruby ${{ matrix.ruby }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [ '2.6', '2.7', '3.0', '3.1', '3.2' ] #this feels like overkill, but we're still on ruby 2.6 in this repo & presumably want to support through 3.2 -- though 3.1 and 3.2 would need alterations to pass
+
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v3
+
+      - name: Install Ruby ${{ matrix.ruby }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+
+      - name: Install dependencies
+        run: bundle install
+
+      - name: Run Rubocop
+        run: bundle exec rubocop
+
+  test:
+    name: Test (Rails ${{ matrix.rails }}, Sidekiq ${{ matrix.sidekiq }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        rails: [ '5.2', '6.0', '6.1', '7.0', '7.1' ]
+        sidekiq: ['5.2', '6.0', '6.1', '6.2', '6.3', '6.4']
```

Comment:
It would be expensive to test this many versions of Rails and Sidekiq against many installs of Ruby as well.

For now I suggest we leave it with just the gem's version of ruby.

Reply:
@kraila because this is an public repository, we don't need to worry about the expense here, [it's free](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#:~:text=GitHub%20Actions%20usage%20is%20free%20for%20standard%20GitHub%2Dhosted%20runners%20in%20public%20repositories%2C%20and%20for%20self%2Dhosted%20runners.)!

What is more important in this case, in my opinion, is ensuring we have a comprehensive matrix of Ruby, Rails, and Sidekiq based on our official support in the gem. Looking at the [gemspec](https://github.com/ezcater/sidekiq_publisher/blob/cd908d3340e610e39fefdddfab5300d9eaadee8c/sidekiq_publisher.gemspec), I think we are pretty spot on, with the exception of missing [Rails 5.1](https://github.com/ezcater/sidekiq_publisher/blob/cd908d3340e610e39fefdddfab5300d9eaadee8c/sidekiq_publisher.gemspec#L63).

To simplify both our GHA workflow and general maintenance, we might consider removing support for old versions of Ruby, Rails, and Sidekiq. Supporting Ruby < 3, Rails < 6.1 (mayyyyybe 6.0), and Sidekiq < 6.0 doesn't seem worthwhile.